app/lib/brexit_checker/action.rb

```diff
@@ -88,8 +88,8 @@ def path_from_url(full_url)
   end
 
   def citizen_action_has_grouping_criteria
-    if audience == "citizen" && grouping_criteria.nil?
-      errors.add(:grouping_criteria, "Can't be empty for citizen actions")
+    if audience == "citizen" && grouping_criteria.blank?
```

Comment:
so what's the deal with this one? Was `grouping_criteira` an array all along?

Reply:
It's an array of criteria keys, ie strings. If the grouping criteria is left off the action on the spreadsheet, when we import the action, `grouping_criteria.nil?` would return true so it works fine. But I changed it to `.blank?` so we would catch `nil`, `[]`, and `{}` so if a dev manually adds new grouping criteria and gets the structure wrong something fails!

Reply:
much better, great improvement!